File: bindings/swift/examples/prepare_merge_coins.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import Foundation
import IotaSDK

@main
struct PrepareMergeCoinsExample {
  static func main() async throws {
    let client = GraphQlClient.newDevnet()

    let sender = try Address.fromHex(
      hex: "0x611830d3641a68f94a690dcc25d1f4b0dac948325ac18f6dd32564371735f32c")

    let coin0 = try PtbArgument.objectIdFromHex(
      hex: "0x0b0270ee9d27da0db09651e5f7338dfa32c7ee6441ccefa1f6e305735bcfc7ab")
    let coin1 = try PtbArgument.objectIdFromHex(
      hex: "0xd04077fe3b6fad13b3d4ed0d535b7ca92afcac8f0f2a0e0925fb9f4f0b30c699")

    let builder = TransactionBuilder(sender: sender).withClient(client: client)

    _ = builder.mergeCoins(primaryCoin: coin0, consumedCoins: [coin1])

    let txn = try await builder.finish()

    print("Signing Digest:", txn.signingDigestHex())
    print("Txn Bytes:", txn.toBase64())

    let res = try await builder.dryRun()
    if res.error != nil {
      throw NSError(
        domain: "PrepareMergeCoins", code: 1,
        userInfo: [NSLocalizedDescriptionKey: "Failed to merge coins: \(res.error!)"])
    }

    print("Merge coins dry run was successful!")
  }
}

File: bindings/wasm/examples/prepare_merge_coins.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Address, GraphQlClient, PtbArgument, initAsync } from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newTestnet();
const sender = Address.fromHex(
  "0xda1820edf693ee32b5729907b9b2ec8e64980ee8c008c17e89cfb4e5ecd72151",
);

const coin0 = PtbArgument.objectIdFromHex(
  "0xdc956de89b914e6a7fbd83caebefc8ec91be1207667ea5576386391aa82449cc",
);
const coin1 = PtbArgument.objectIdFromHex(
  "0x65beb18e282d1f33a39bffa84ff92ec4d2fec0350ba6f7e5a568afff72d651db",
);

const builder = client.transactionBuilder(sender);
builder.mergeCoins(coin0, [coin1]);

const txn = await builder.finish();
console.log("Signing Digest:", txn.signingDigestHex());
console.log("Txn Bytes:", txn.toBase64());

const res = await builder.dryRun();
if (res.error) {
  throw new Error(`Failed to merge coins: ${res.error}`);
}
console.log("Merge coins dry run was successful!");
